controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -86,6 +86,11 @@ func (w *Workload) UpdateCoreDNS(ctx context.Context, kcp *controlplanev1.Kubead
 		return nil
 	}
 
+	// Return early if DNS is disabled
+	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.DNS.Disabled {
+		return nil
+	}
+
```

Comment:
```suggestion
	// Return early if DNS is disabled
	if clusterCondiguration.DNS.Disabled {
		return nil
	}

```

We can rewrite it like suggestion when you move these codes after L-94,